### DISCUSSION TITLE
Mongoose 6.11.4 on lambda

### DISCUSSION THREAD

**Discussion by jmontg03** [Automattic/mongoose#13852] [Q&A]
Hi,
I have an nextjs server running on a lambda, which is running [payloadcms](https://github.com/payloadcms). I have identified that it seems like mongoose is returning the lambda somehow on a Model.FindOne(). Heres the snippet im running, which is an adjustment of - https://github.com/payloadcms/payload/blob/master/src/auth/operations/login.ts#L84 where i see the issue.

I am connecting to a mongodb atlas serverless cluster, although, as payload handles the actual connection, i dont have the initialisation options being selected, however, we have observed its ability to connect and write to the database. Perhaps the connection is being held mistakenly by the lambda as it seems its opened on startup?

Atlas is configured to allow all IPs and is authorising through username and password (aware this is a risk, wont be done in prod)

```
console.log('Fetching from mongoose')
try {
    console.log(await Model.findOne({ email:'test@test.com' }).lean());
} catch (error) {
    console.error(error);
}
console.log('Should get here')
```

The behaviour we are seeing is 'Fetching from mongoose' is outputting, as expected, followed by the lambda terminating. There is no error message caught from here. I have checked memory usage, the lambda as 2gb assigned, uses little over 300mb, doesnt seem to be a leak anywhere. 

Any ideas are appreciated!

**Comment by vkarpov15** [comment]
Have you actually connected to MongoDB? Try printing out `Model.db.readyState` in `console.log('Fetching from mongoose')`
